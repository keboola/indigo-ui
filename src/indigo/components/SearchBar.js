import React from 'react';
import { FormControl } from 'react-bootstrap';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Icon from './icons';

class SearchBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      query: this.props.query,
    };
    this.handleKeyDown = this.handleKeyDown.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleKeyDown(event) {
    this.props.onKeyDown(event.key);
  }

  handleChange(event) {
    this.setState({ query: event.target.value });
    this.props.onChange(event.target.value);
  }

  handleSubmit(event) {
    event.preventDefault();
    this.props.onSubmit();
  }

  renderRecommendedLink(linkElement) {
    return (
      <span className="searchbar-recommended-link">
        <Icon.Search className="searchbar-recommended-link-icon icon-size-16" />
        {linkElement}
      </span>
    );
  }

  render() {
    return (
      <div
        className={classNames(
          'searchbar',
          {
            [`searchbar-${this.props.theme}`]: !!this.props.theme,
          },
          this.props.className
        )}
      >
        <form onSubmit={this.handleSubmit} className="searchbar-form">
          <FormControl
            value={this.state.query}
            placeholder={this.props.placeholder}
            onChange={this.handleChange}
            onKeyDown={this.handleKeyDown}
            className="searchbar-input"
            type="text"
            autoFocus
            inputRef={this.props.inputRef}
          />
          <Icon.Search className="searchbar-icon icon-size-20" />
        </form>
        {this.props.additionalActions && (
          <div className="searchbar-actions">{this.props.additionalActions}</div>
        )}
        {this.props.recommendedSearches && (
          <div className="searchbar-recommended-link-list">
            {this.props.recommendedSearches
              .map((link) => this.renderRecommendedLink(link))
              .reduce((prev, curr) => [prev, ', ', curr])}
          </div>
        )}
      </div>
    );
  }
}

SearchBar.defaultProps = {
  onKeyDown: () => {},
  onChange: () => {},
  onSubmit: () => {},
  placeholder: 'Search',
  query: '',
};

SearchBar.propTypes = {
  query: PropTypes.string,
  onChange: PropTypes.func,
  onKeyDown: PropTypes.func,
  onSubmit: PropTypes.func,
  inputRef: PropTypes.func,
  className: PropTypes.string,
  placeholder: PropTypes.string,
  additionalActions: PropTypes.oneOfType([PropTypes.array, PropTypes.element]),
  recommendedSearches: PropTypes.oneOfType([PropTypes.array, PropTypes.element]),
  theme: PropTypes.oneOf(['inverse']),
};

export default SearchBar;
