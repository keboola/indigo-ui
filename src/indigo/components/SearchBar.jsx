import React from 'react';
import { FormControl } from 'react-bootstrap';
import Icon from './icons'
import PropTypes from 'prop-types';
import classNames from "classnames";

class SearchBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {query: this.props.query};
    this.handleKeyDown = this.handleKeyDown.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  
  handleKeyDown(event) {
    this.setState({keyDown: event.key});
    this.props.onKeyDown(event.key);
  }
  
  handleChange(event) {
    this.setState({query: event.target.value});
    this.props.onChange(event.target.value);
  }
  
  handleSubmit(event) {
    event.preventDefault();
    this.props.onSubmit(this.state.query);
  }
  
  render() {
    return (
      <form
        className={classNames("searchbar", this.props.className)}
        onSubmit={this.handleSubmit}
      >
        <FormControl
          value={this.state.query}
          placeholder={this.props.placeholder}
          ref={this.props.ref}
          onChange={this.handleChange}
          onKeyDown={this.handleKeyDown}
          className="searchbar-input"
          type="text"
          autoFocus
        />
        <Icon.Search className="searchbar-icon icon-size-20" />
        {!!this.props.additionalActions &&
          this.props.additionalActions
        }
      </form>
    );
  }
}

SearchBar.defaultProps = {
  onKeyDown: () => {},
  onChange: () => {},
  onSubmit: () => {},
  placeholder: 'Search'
};

SearchBar.propTypes = {
  ref: PropTypes.string,
  query: PropTypes.string,
  onChange: PropTypes.func,
  onKeyDown: PropTypes.func,
  onSubmit: PropTypes.func,
  className: PropTypes.string,
  placeholder: PropTypes.string,
  additionalActions: PropTypes.object
};

export default SearchBar;
