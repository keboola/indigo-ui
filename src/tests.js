import renderer from 'react-test-renderer';

export const snapshot = (component) => {
  const tree = renderer
    .create(component)
    .toJSON();
  expect(tree).toMatchSnapshot();
};
