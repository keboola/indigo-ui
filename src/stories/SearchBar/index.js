import React from 'react';
import {storiesOf} from '@storybook/react';
import {withInfo} from '@storybook/addon-info';
import {Button} from 'react-bootstrap';
import {Icon} from '../../indigo/components';
import { action } from '@storybook/addon-actions';

import SearchBar from '../../indigo/components/SearchBar';


const demoContent = (
  <Button bsStyle="default">
    <Icon.Help className={'icon-size-20'}/> Demo Button
  </Button>
)
storiesOf('SearchBar', module)
  .add(
    'Basic usage',
    withInfo({
      text: `
        usage: ex, tr, wr, orch, versions
        `,
      inline: true,
    })(() => {
      return (
        <SearchBar/>
      );
    })
  )
  .add(
    'Searchbar with actions',
    withInfo({
      text: `
        usage: config rows, config detail
        `,
      inline: true,
    })(() => {
      return (
        <SearchBar actions={demoContent}/>
      );
    })
  )
  
  .add(
    'Custom placeholder',
    withInfo({
      text: `
        usage: ex, tr, wr, orch, versions
        `,
      inline: true,
    })(() => {
      return (
        <SearchBar
          placeholder="Search by component name, desription or id"
        />
      );
    })
  )
  .add(
    'Events demo',
    withInfo({
      text: `
        onChange & onSubmit demo
        `,
      inline: true,
    })(() => {
      return (
        <SearchBar
          onChange={action('changed content')}
          onSubmit={action('submitted')}
        />
      );
    })
  )
  .add(
    'Prefilled query',
    withInfo({
      text: `
        Prefilled query
        `,
      inline: true,
    })(() => {
      return (
        <SearchBar
          query="Look ma` I am prefilled"
        />
      );
    })
  )
  .add(
    'Select project',
    withInfo({
      text: `
        to be done
        `,
      inline: true,
    })(() => {
      return (
        <ul className="list-unstyled">
          <li className="dropdown-header kb-nav-search kbc-search">
            <span className="kbc-icon-search"></span>
            <input className="form-control" placeholder="Search your projects" value=""/>
          </li>
        </ul>
      );
    })
  );
