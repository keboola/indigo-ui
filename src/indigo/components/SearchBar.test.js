import React from 'react';

import { matchesSnapshot } from '../../tests';
import SearchBar from './SearchBar';

describe('<SearchBar />', () => {
  it('SearchBar', () => {
    matchesSnapshot(<SearchBar onChange={() => null} />);
  });

  it('SearchBar with autofocus', () => {
    matchesSnapshot(<SearchBar autoFocus onChange={() => null} />);
  });

  it('SearchBar with border', () => {
    matchesSnapshot(<SearchBar onChange={() => null} bordered />);
  });

  it('SearchBar - with one additional button', () => {
    matchesSnapshot(
      <SearchBar
        onChange={() => null}
        additionalActions={<button className="btn btn-default">Submit</button>}
      />
    );
  });

  it('SearchBar - with two additional buttons', () => {
    matchesSnapshot(
      <SearchBar
        onChange={() => null}
        additionalActions={[
          <button key="submit-button" className="btn btn-primary">
            Submit
          </button>,
          <button key="reset-button" className="btn btn-default">
            Reset
          </button>,
        ]}
      />
    );
  });
});
