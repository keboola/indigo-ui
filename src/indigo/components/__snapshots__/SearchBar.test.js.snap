// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> SearchBar - inverse theme 1`] = `
<div
  className="searchbar searchbar-inverse"
>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      className="searchbar-input form-control"
      id={undefined}
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
    <svg
      className="icon searchbar-icon icon-size-20"
      height="32"
      viewBox="0 0 18 18"
      width="32"
    >
      <path
        d="M12.5 11h-.79l-.28-.27A6.471 6.471 0 0 0 13 6.5 6.5 6.5 0 1 0 6.5 13c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L17.49 16l-4.99-5zm-6 0C4.01 11 2 8.99 2 6.5S4.01 2 6.5 2 11 4.01 11 6.5 8.99 11 6.5 11z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </form>
</div>
`;

exports[`<SearchBar /> SearchBar - with one additional button 1`] = `
<div
  className="searchbar"
>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      className="searchbar-input form-control"
      id={undefined}
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
    <svg
      className="icon searchbar-icon icon-size-20"
      height="32"
      viewBox="0 0 18 18"
      width="32"
    >
      <path
        d="M12.5 11h-.79l-.28-.27A6.471 6.471 0 0 0 13 6.5 6.5 6.5 0 1 0 6.5 13c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L17.49 16l-4.99-5zm-6 0C4.01 11 2 8.99 2 6.5S4.01 2 6.5 2 11 4.01 11 6.5 8.99 11 6.5 11z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </form>
  <div
    className="searchbar-actions"
  >
    <button
      className="btn btn-default"
    >
      Submit
    </button>
  </div>
</div>
`;

exports[`<SearchBar /> SearchBar - with two additional buttons 1`] = `
<div
  className="searchbar"
>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      className="searchbar-input form-control"
      id={undefined}
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
    <svg
      className="icon searchbar-icon icon-size-20"
      height="32"
      viewBox="0 0 18 18"
      width="32"
    >
      <path
        d="M12.5 11h-.79l-.28-.27A6.471 6.471 0 0 0 13 6.5 6.5 6.5 0 1 0 6.5 13c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L17.49 16l-4.99-5zm-6 0C4.01 11 2 8.99 2 6.5S4.01 2 6.5 2 11 4.01 11 6.5 8.99 11 6.5 11z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </form>
  <div
    className="searchbar-actions"
  >
    <button
      className="btn btn-primary"
    >
      Submit
    </button>
    <button
      className="btn btn-default"
    >
      Reset
    </button>
  </div>
</div>
`;

exports[`<SearchBar /> SearchBar 1`] = `
<div
  className="searchbar"
>
  <form
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      className="searchbar-input form-control"
      id={undefined}
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
    <svg
      className="icon searchbar-icon icon-size-20"
      height="32"
      viewBox="0 0 18 18"
      width="32"
    >
      <path
        d="M12.5 11h-.79l-.28-.27A6.471 6.471 0 0 0 13 6.5 6.5 6.5 0 1 0 6.5 13c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L17.49 16l-4.99-5zm-6 0C4.01 11 2 8.99 2 6.5S4.01 2 6.5 2 11 4.01 11 6.5 8.99 11 6.5 11z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
  </form>
</div>
`;
